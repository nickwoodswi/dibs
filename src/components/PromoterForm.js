import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { Route } from 'react-router-dom'
import StartDateSelector from './StartDateSelector'
import EndDateSelector from './EndDateSelector'
import MultiHoldSelector from './MultiHoldSelector'
import SingleHold from './SingleHold'
import './PromoterForm.css'

class PromoterForm extends Component {
  render() {
    return(
      <div className="promoter-form">
        <div className="hold-type-header">PLACING HOLD(S) ON:</div>
        <div className="hold-type-buttons">
          <Link to="/promoter/single"><button id="single" name="hold_type" value="single" onClick={e => this.props.handleState('hold_type', e.target.value)}>A SINGLE DATE</button></Link>
          <Link to="/promoter/range"><button id="range" name="hold_type" value="range" onClick={e => this.props.handleState('hold_type', e.target.value)}>A RANGE OF DATES</button></Link>
          <Link to="/promoter/multi"><button id="multiple" name="hold_type" value="multiple" onClick={e => this.props.handleState('hold_type', e.target.value)}>MULTIPLE DATES</button></Link>
        </div>
        <Route path="/promoter/single">
          <SingleHold 
            appState={this.props.appState} 
            handleSingleHold={this.props.handleSingleHold}
            submitForm={this.props.submitForm} />
        </Route>
        <Route path="/promoter/range">
          <div className="range-date-selector">
            <StartDateSelector 
              appState={this.props.appState} 
              handleState={this.props.handleState} 
              submitForm={this.props.submitForm}/><br/>
            <EndDateSelector 
              appState={this.props.appState} 
              handleState={this.props.handleState} />
          </div>
        </Route>
        <Route path="/promoter/multi">
            <MultiHoldSelector 
              appState={this.props.appState} 
              handleState={this.props.handleState} 
              addHold={this.props.addHold}
              deleteHold={this.props.deleteHold}
              submitForm={this.props.submitForm}/>
        </Route>
        <div className="performance-type-detail">
          FOR:
            <input type="text" placeholder="Band, Artist, or Performer Title/Name" value={this.props.appState.contact_act} onChange={e => this.props.handleState('contact_act', e.target.value)} />
            <input type="text" placeholder="Your Home ZIP Code" value={this.props.appState.org_zip} onChange={e => this.props.handleState('org_zip', e.target.value)} />
            <input type="radio" id="band" name="performance_type" value="band" onChange={e => this.props.handleState('event_type', e.target.value)} />Band
            <input type="radio" id="solo" name="performance_type" value="solo" onChange={e => this.props.handleState('event_type', e.target.value)} />Solo Performer
            <input type="radio" id="dj" name="performance_type" value="dj" onChange={e => this.props.handleState('event_type', e.target.value)} />DJ
        </div>

        <div className="media">
          Please link to the headlining artist's music, or a look at their performance:
            <input type="text" value={this.props.appState.spotify} placeholder="Spotify Artist ID" onChange={e => this.props.handleState('spotify', e.target.value)} />
            <input type="text" value={this.props.appState.youtube} placeholder="YouTube Channel or Video" onChange={e => this.props.handleState('youtube', e.target.value)} />
            <input type="text" value={this.props.appState.facebook} placeholder="Facebook Page" onChange={e => this.props.handleState('facebook', e.target.value)} />
            <input type="text" value={this.props.appState.bandcamp} placeholder="Bandcamp" onChange={e => this.props.handleState('bandcamp', e.target.value)} />
            <input type="text" value={this.props.appState.website} placeholder="Website" onChange={e => this.props.handleState('website', e.target.value)} />
        </div>

        <div className="support-need">
          Do you need us to book local support?
            <input type="radio" id="support_need" name="support_need" value="yes" onChange={e => this.props.handleState('support_need', e.target.value)} />Yes
            <input type="radio" id="support_need" name="support_need" value="no" onChange={e => this.props.handleState('support_need', e.target.value)} />No
        </div>

          <div className="package-support-header">PACKAGE SUPPORT</div>

          <div className="performance-type-detail">
            <input type="text" onChange={e => this.props.handleState('dsupport_title', e.target.value)} value={this.props.appState.dsupport_title} placeholder="Direct Support" />
            <input onChange={e => this.props.handleState('dsupport_type', e.target.value)} type="radio" id="band-2" name="support_type-2" value="band" />Band
            <input onChange={e => this.props.handleState('dsupport_type', e.target.value)} type="radio" id="solo-2" name="support_type-2" value="solo" />Solo Performer
            <input onChange={e => this.props.handleState('dsupport_type', e.target.value)} type="radio" id="dj-2" name="support_type-2" value="dj" />DJ
          </div>

          <div className="media">
            <input type="text" onChange={e => this.props.handleState('spotify2', e.target.value)} value={this.props.appState.spotify2} placeholder="Spotify Artist ID" />
            <input type="text" onChange={e => this.props.handleState('youtube2', e.target.value)} value={this.props.appState.youtube2} placeholder="YouTube Video or Channel" />
            <input type="text" onChange={e => this.props.handleState('facebook2', e.target.value)} value={this.props.appState.facebook2} placeholder="Facebook Page" />
            <input type="text" onChange={e => this.props.handleState('bandcamp2', e.target.value)} value={this.props.appState.bandcamp2} placeholder="Bandcamp" />
            <input type="text" onChange={e => this.props.handleState('website2', e.target.value)} value={this.props.appState.website2} placeholder="Website" />
          </div>

          <div className="performance-type-detail">
            <input type="text" onChange={e => this.props.handleState('opener_title', e.target.value)} value={this.props.appState.opener_title} placeholder="Opener" />
            <input onChange={e => this.props.handleState('support_type3', e.target.value)} type="radio" id="band-3" name="support_type3" value="band" />Band
            <input onChange={e => this.props.handleState('support_type3', e.target.value)} type="radio" id="solo-3" name="support_type3" value="solo" />Solo Performer
            <input onChange={e => this.props.handleState('support_type3', e.target.value)} type="radio" id="dj-3" name="support_type3" value="dj" />DJ
          </div>

          <div className="media">
            <input type="text" onChange={e => this.props.handleState('spotify3', e.target.value)} value={this.props.appState.spotify3} placeholder="Spotify Artist ID" /><br/>
            <input type="text" onChange={e => this.props.handleState('youtube3', e.target.value)} value={this.props.appState.youtube3} placeholder="Youtube Video or Channel" /><br/>
            <input type="text" onChange={e => this.props.handleState('facebook3', e.target.value)} value={this.props.appState.facebook3} placeholder="Facebook Page" /><br/>
            <input type="text" onChange={e => this.props.handleState('bandcamp3', e.target.value)} value={this.props.appState.bandcamp3} placeholder="Bandcamp" /><br/>
            <input type="text" onChange={e => this.props.handleState('website3', e.target.value)} value={this.props.appState.website3} placeholder="Website" /><br/>
          </div>

          <div className="performance-type-detail">
            <input type="text" onChange={e => this.props.handleState('opener2_title', e.target.value)} value={this.props.appState.opener2_title} placeholder="Second Opener Title" />
            <input type="text" onChange={e => this.props.handleState('support_type4', e.target.value)} type="radio" id="band-4" name="support_type4" value="band" />Band
            <input type="text" onChange={e => this.props.handleState('support_type4', e.target.value)} type="radio" id="solo-4" name="support_type4" value="solo" />Solo Performer
            <input type="text" onChange={e => this.props.handleState('support_type4', e.target.value)} type="radio" id="dj-4" name="support_type4" value="dj" />DJ
          </div>

          <div className="media">
            <input type="text" onChange={e => this.props.handleState('spotify4', e.target.value)} value={this.props.appState.spotify4} placeholder="Spotify Artist ID" /><br/>
            <input type="text" onChange={e => this.props.handleState('youtube4', e.target.value)} value={this.props.appState.youtube4} placeholder="YouTube Video or Channel" /><br/>
            <input type="text" onChange={e => this.props.handleState('facebook4', e.target.value)} value={this.props.appState.facebook4} placeholder="Facebook Page" /><br/>
            <input type="text" onChange={e => this.props.handleState('bandcamp4', e.target.value)} value={this.props.appState.bandcamp4} placeholder="Bandcamp" /><br/>
            <input type="text" onChange={e => this.props.handleState('website4', e.target.value)} value={this.props.appState.website4} placeholder="Website" /><br/>
          </div>
          

        <div className="event-detail">
          Provide any specific technical needs or details here:
          <input id="event-detail" type="text" onChange={e => this.props.handleState('event_detail', e.target.value)} value={this.props.appState.event_detail} />
        </div>

        <div className="org-detail">
          Your Organization: <input type="text" onChange={e => this.props.handleState('contact_org', e.target.value)} value={this.props.appState.contact_org} placeholder="Name" />
          Phone: <input type="text" onChange={e => this.props.handleState('contact_phone', e.target.value)} value={this.props.appState.contact_phone} placeholder="###" /><br/><br/>
          Mailing Address:
          <input type="text" onChange={e => this.props.handleState('org_street1', e.target.value)} value={this.props.appState.org_street1} placeholder="Street 1" />
          <input type="text" onChange={e => this.props.handleState('org_street2', e.target.value)} value={this.props.appState.org_street2} placeholder="Street 2" />
          <input type="text" onChange={e => this.props.handleState('org_city', e.target.value)} value={this.props.appState.org_city} placeholder="City" />
          <input type="text" onChange={e => this.props.handleState('org_country', e.target.value)} value={this.props.appState.org_country} placeholder="Country" />
          <input type="text" onChange={e => this.props.handleState('org_zip', e.target.value)} value={this.props.appState.org_zip} placeholder="Postal Code" />
        </div>

        <div className="place-hold-button-container">
          <Link to="/holdconfirmation"><button onClick={e => this.props.submitForm()}>PLACE HOLDS</button></Link>
        </div>

      </div>
    )
}
}

export default PromoterForm