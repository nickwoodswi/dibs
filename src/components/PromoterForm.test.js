import React from 'react';
import { render } from '@testing-library/react';
import PromoterForm from './AgentForm';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders promoter form', () => {
    render(<MemoryRouter><PromoterForm /></MemoryRouter>);
});

it('PromoterForm renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><PromoterForm /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});