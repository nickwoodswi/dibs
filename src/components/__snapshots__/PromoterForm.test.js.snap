// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoterForm renders the UI as expected 1`] = `
<div
  className="agent-form"
>
  <div
    className="hold-type-header"
  >
    PLACING HOLD(S) ON:
  </div>
  <div
    className="hold-type-buttons"
  >
    <a
      aria-current={null}
      href="/agent/single"
      onClick={[Function]}
    >
      <div
        className="hold_type"
        id="single"
        onClick={[Function]}
        value="single"
      >
        A SINGLE DATE
      </div>
    </a>
    <a
      aria-current={null}
      href="/agent/range"
      onClick={[Function]}
    >
      <div
        className="hold_type"
        id="range"
        onClick={[Function]}
        value="range"
      >
        A RANGE OF DATES
      </div>
    </a>
    <a
      aria-current={null}
      href="/agent/multi"
      onClick={[Function]}
    >
      <div
        className="hold_type"
        id="multiple"
        onClick={[Function]}
        value="multiple"
      >
        MULTIPLE DATES
      </div>
    </a>
  </div>
  <div
    className="performance-type-detail"
  >
    FOR:
    <input
      onChange={[Function]}
      placeholder="Band, Artist, or Performer Title/Name"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Your Home ZIP Code"
      type="text"
    />
    <div
      className="performer-type"
    >
      <input
        id="band"
        name="performance_type"
        onChange={[Function]}
        type="radio"
        value="band"
      />
      Band
      <input
        id="solo"
        name="performance_type"
        onChange={[Function]}
        type="radio"
        value="solo"
      />
      Solo Performer
      <input
        id="dj"
        name="performance_type"
        onChange={[Function]}
        type="radio"
        value="dj"
      />
      DJ
    </div>
  </div>
  <div
    className="media"
  >
    Please link to your client's music, or a look at their performance:
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="YouTube Channel or Video"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
  </div>
  <div
    className="support-need"
  >
    Do you need us to book local support?
    <input
      id="support_need"
      name="support_need"
      onChange={[Function]}
      type="radio"
      value="yes"
    />
    Yes
    <input
      id="support_need"
      name="support_need"
      onChange={[Function]}
      type="radio"
      value="no"
    />
    No
  </div>
  <div
    className="package-support-header"
  >
    PACKAGE SUPPORT
  </div>
  <div
    className="performance-type-detail"
  >
    <input
      onChange={[Function]}
      placeholder="Direct Support"
      type="text"
    />
    <div
      className="performer-type"
    >
      <input
        id="band-2"
        name="support_type-2"
        onChange={[Function]}
        type="radio"
        value="band"
      />
      Band
      <input
        id="solo-2"
        name="support_type-2"
        onChange={[Function]}
        type="radio"
        value="solo"
      />
      Solo Performer
      <input
        id="dj-2"
        name="support_type-2"
        onChange={[Function]}
        type="radio"
        value="dj"
      />
      DJ
    </div>
  </div>
  <div
    className="media"
  >
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="YouTube Video or Channel"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
  </div>
  <div
    className="performance-type-detail"
  >
    <input
      onChange={[Function]}
      placeholder="Opener"
      type="text"
    />
    <div
      className="performer-type"
    >
      <input
        id="band-3"
        name="support_type3"
        onChange={[Function]}
        type="radio"
        value="band"
      />
      Band
      <input
        id="solo-3"
        name="support_type3"
        onChange={[Function]}
        type="radio"
        value="solo"
      />
      Solo Performer
      <input
        id="dj-3"
        name="support_type3"
        onChange={[Function]}
        type="radio"
        value="dj"
      />
      DJ
    </div>
  </div>
  <div
    className="media"
  >
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Youtube Video or Channel"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
    <br />
  </div>
  <div
    className="performance-type-detail"
  >
    <input
      onChange={[Function]}
      placeholder="Second Opener"
      type="text"
    />
    <div
      className="performer-type"
    >
      <input
        id="band-4"
        name="support_type4"
        onChange={[Function]}
        type="radio"
        value="band"
      />
      Band
      <input
        id="solo-4"
        name="support_type4"
        onChange={[Function]}
        type="radio"
        value="solo"
      />
      Solo Performer
      <input
        id="dj-4"
        name="support_type4"
        onChange={[Function]}
        type="radio"
        value="dj"
      />
      DJ
    </div>
  </div>
  <div
    className="media"
  >
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="YouTube Video or Channel"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
    <br />
  </div>
  <div
    className="event-detail"
  >
    Provide any specific technical needs or details here:
    <textarea
      id="event-detail"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="org-detail"
  >
    Agency: 
    <input
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    Phone: 
    <input
      onChange={[Function]}
      placeholder="###"
      type="text"
    />
    <br />
    <br />
    Mailing Address:
    <input
      onChange={[Function]}
      placeholder="Street 1"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Street 2"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="City"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Country"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Postal Code"
      type="text"
    />
  </div>
  <div
    className="place-hold-button-container"
  >
    <a
      aria-current={null}
      href="/holdconfirmation"
      onClick={[Function]}
    >
      <div
        className="submit-button"
        onClick={[Function]}
      >
        +PLACE HOLDS
      </div>
    </a>
  </div>
</div>
`;
